### ISSUE TITLE
Wrong editor default

### ISSUE THREAD

**Issue by omangin** [pubs/pubs#46]
The default `edit_cmd` in the config is `'None'`. Instead of failing to open the command `None`, the value of `$EDITOR` should be used, with some sane default (*vi*).

**Comment by humm** [comment]
H

**Comment by omangin** [comment]
Do we support window?

Actually `edit_cmd` was `'None'` on my computer from the config the updater generated. So something is at least not working.

Also, my opinion is that the user should not need to have any `edit_cmd` defined in the config for the expected thing to happen. Also if the user changes its `$editor`, why should he also change it in every application he uses that may call an editor?

It is great to try *very hard* to find a working editor, but is it necessary? Most users who care about an editor have it specified as `$EDITOR` or would set it in the config after their first edit. One could argue that newbies would be unable to get out of `vi` if used as default. Most of these users are using Ubuntu, that, if I am correct, `export EDITOR=nano` in the default `bashrc` so this won't happen. I do not know what happens on MacOS. Also the order for `[vim, nano, emacs, vi]` seems quite arbitrary.

In summary, I think we should favor convention over configuration. Which the code seems to attempt, but in my opinion in a wrong way, by hard-coding the convention in the configuration at init time. In my opinion this adds not-really-useful code to be maintained for a functionality that is already implemented  by the environment variables.

So by order of priority, I think we should support the configuration not being set. Then we may try some magic guess but in my opinion this is useless.

**Comment by humm** [comment]
You raise very good points. You are indeed right that we should support the configuration not being set. And windows support is probably a quagmire of undue proportions.

On MacOS X, EDITOR is not set, and I do think nano is a better option than vi for users that do not have strong opinions about editors. I removed my overly complicated code and made nano the default.

**Comment by omangin** [comment]
OK. Thanks. There is still one minor issue which is that I am not sure nano is always installed on UNIX systems. That is why `vi` is a more secure default since it is required by POSIX.  I however agree that it is better to default to nano for inexperienced users.

So the options are:
- keep using `nano` as `pubs` default, knowing that in theory it is not guaranteed to work (although it would on most platforms),
- use `vi` which will always work, assuming that distributions that target inexperienced users (like Ubuntu) would have take care of setting an appropriate `$EDITOR` which would be found by `pubs`.

I understand that this is more a matter of principle w.r.t. standards than a practical issue…

**Comment by humm** [comment]
Valid point. I am gonna let you decide this one :P
